### PR TITLE
aws_array_list_erase()

### PR DESCRIPTION
**Pull request by graebm**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/common/array_list.inl [hunk 1]
```diff
@@ -169,6 +169,37 @@ void aws_array_list_pop_front_n(struct aws_array_list *AWS_RESTRICT list, size_t
     AWS_POSTCONDITION(aws_array_list_is_valid(list));
 }
 
+int aws_array_list_erase(struct aws_array_list *AWS_RESTRICT list, size_t index) {
+    AWS_PRECONDITION(aws_array_list_is_valid(list));
+
+    size_t length  = aws_array_list_length(list);
```
Comment (justinboswell) [inline include/aws/common/array_list.inl]: const

File: include/aws/common/array_list.inl [hunk 2]
```diff
@@ -169,6 +169,37 @@ void aws_array_list_pop_front_n(struct aws_array_list *AWS_RESTRICT list, size_t
...
+
+    if (index >= length) {
+        AWS_POSTCONDITION(aws_array_list_is_valid(list));
+        return aws_raise_error(AWS_ERROR_INVALID_INDEX);
+    }
+
+    if (index == 0) {
+        /* Removing front element */
+        aws_array_list_pop_front(list);
+    } else if (index == length - 1) {
```
Comment (justinboswell) [inline include/aws/common/array_list.inl]: (length - 1)

File: include/aws/common/array_list.inl (hunk 1 above)
Comment (graebm) [reply include/aws/common/array_list.inl]: did

File: include/aws/common/array_list.inl (hunk 2 above)
Comment (graebm) [reply include/aws/common/array_list.inl]: did

**Pull request merged**